src/hooks/user-list.hook.ts

```diff
@@ -0,0 +1,84 @@
+import {useState} from 'react';
+import {useQuery, gql} from '@apollo/client';
+
+const GET_USERS = gql`
+  query GetUsers($data: PageInput!) {
+    users(data: $data) {
+      nodes {
+        id
+        name
+        email
+      }
+      count
+      pageInfo {
+        hasNextPage
+      }
+    }
+  }
+`;
+
+type User = {
+  id: string;
+  name: string;
+  email: string;
+};
```

Comment:
```suggestion
interface User {
  id: string;
  name: string;
  email: string;
}
```

Por convenção, normalmente usamos interface. E também elas são mais fáceis de compor outros tipos, pq tem `extends`.